controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -815,6 +836,11 @@ func (r *BareMetalHostReconciler) actionPreparing(prov provisioner.Provisioner,
 		return result
 	}
 
+	// store settings updated after cleaning
+	if err = r.storeHostFirmwareSettings(prov, info); err != nil {
```

Comment:
The `storeHostFirmwareSettings` does a lot of things, is it fully required here?